src/NServiceBus.AzureStoragePersistence/Timeout/TimeoutLogic/TimeoutPersister.cs

```diff
@@ -85,27 +85,29 @@ public TimeoutPersister(string timeoutConnectionString, string timeoutDataTableN
             var lastSuccessfulReadEntity = GetLastSuccessfulRead(timeoutManagerDataTable);
             var lastSuccessfulRead = lastSuccessfulReadEntity?.LastSuccessfullRead;
 
-            IQueryable<TimeoutDataEntity> query;
+            TableQuery<TimeoutDataEntity> query;
 
             if (lastSuccessfulRead.HasValue)
             {
-                query = from c in timeoutDataTable.CreateQuery<TimeoutDataEntity>()
-                        where string.Compare(c.PartitionKey, lastSuccessfulRead.Value.ToString(partitionKeyScope), StringComparison.InvariantCultureIgnoreCase) != 0
-                            && string.Compare(c.PartitionKey, now.ToString(partitionKeyScope), StringComparison.InvariantCultureIgnoreCase) != 0
-                            && c.OwningTimeoutManager == endpointName
-                        select c;
+                query = new TableQuery<TimeoutDataEntity>()
```

Comment:
Information only comment.
This query will issue table scans and as it's the main query behind dispatching timeouts it'll be used a lot. The porting from LINQ to the `TableQuery` filters is done properly. I've raised a separate issue to consider performance improvements we can make in timeouts https://github.com/Particular/NServiceBus.AzureStoragePersistence/issues/46